reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/opentracing/AndroidTracerTest.kt

```diff
@@ -453,14 +459,18 @@ class AndroidTracerTest {
         assertThat(childSpanId).isNotEqualTo(spanId)
         val eventsWritten = stubSdkCore.eventsWritten(Feature.TRACING_FEATURE_NAME)
         assertThat(eventsWritten).hasSize(2)
-        val event0 = JsonParser.parseString(eventsWritten[0].eventData) as JsonObject
-        val event1 = JsonParser.parseString(eventsWritten[1].eventData) as JsonObject
-        assertThat(event0.getString("env")).isEqualTo(stubSdkCore.getDatadogContext().env)
-        assertThat(event0.getString("spans[0].trace_id")).isEqualTo(traceId)
-        assertThat(event0.getString("spans[0].span_id")).isEqualTo(spanId)
-        assertThat(event1.getString("env")).isEqualTo(stubSdkCore.getDatadogContext().env)
-        assertThat(event1.getString("spans[0].trace_id")).isEqualTo(childTraceId)
-        assertThat(event1.getString("spans[0].span_id")).isEqualTo(childSpanId)
+        val payload0 = JsonParser.parseString(eventsWritten[0].eventData) as JsonObject
+        val payload1 = JsonParser.parseString(eventsWritten[1].eventData) as JsonObject
+        PayloadAssert.assertThat(payload0).hasChildAt(0) {
+            hasEnv(stubSdkCore.getDatadogContext().env)
+            hasTraceId(traceId)
+            hasSpanId(spanId)
+        }
+        PayloadAssert.assertThat(payload1).hasChildAt(0) {
```

Comment:
and here is an example of confusing API. In reality `payload1` was at index 1

Reply:
hmm this should have failed, I will look into it

Reply:
not exactly, the payloads contain multiple spans, in these tests we are asserting only the `span[0]` but in the Otel tests I am planning to test also a scope which can contain multiple spans in a trace (payload). This test is fine as it asserts the first span in `payload0` and `payload1`